src/DataProtection/README.md

```diff
@@ -1,8 +1,35 @@
-DataProtection
-==============
+# DataProtection
 
 Data Protection APIs for protecting and unprotecting data. You can find documentation for Data Protection in the [ASP.NET Core Documentation](https://docs.microsoft.com/aspnet/core/security/data-protection/).
 
+## Description
+
+The following contains a description of each sub-directory in the `DataProtection` directory.
+
+- `Abstractions`: Contains the source files for the main DataProtection interfaces like IDataProtector and IDataProtectionProvider
+- `Cryptography.Internal`: Contains the source files for cryptography infrastucture, applications and libraries should not reference this package directly.
+- `Cryptography.KeyDerivation`: Contains the source files related to key derivation, i.e. PBKDF2
+- `DataProtection`: Contains the main implementation of DataProtection for ASP.NET Core to protect and unprotect data.
+- `EntityFrameworkCore`: Contains the implementation for storing data using EntityFrameworkCore
+- `Extensions`: Contains a additional apis via extension methods.
+- `StackExchangeRedis`: Contains the implementation for storing data using StackExchangeREdis
+- `samples`: Contains a collection of sample apps
```

Comment:
Are the samples helpful at all in the dev loop?

Reply:
Probably not, were we not supposed to just document every directory?

Reply:
Correct, I was just wondering if there was something helpful outside of running the tests that devs might need to know when making changes in the repo.

Reply:
Ah gotcha, I mistakenly thought you were implying that we shouldn't mention it...  No these appear to just be customer facing samples/examples for commonly asked scenarios, useful for linking.  I don't believe we build samples, but I think that's true generally, so they are all vulnerable to breaking changes causing samples to no longer work...